### DISCUSSION TITLE
supportable SemanticTokenType at Scala 2

### DISCUSSION THREAD

**Discussion by ShintaroSasaki** [ShintaroSasaki/metals#6] [Q&A] [answered]
For current , the following types are server-capable  as default. 
But actually, checked ones will be only provided.  Does it seem appropriate ?
There is  a brief description of each type in the middle of [this page](https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide).
- [x] Type
- [x] Class
- [ ] Enum
- [x] Interface
- [ ] Struct
- [x] TypeParameter
- [x] Parameter
- [x] Variable
- [ ] Property
- [ ] EnumMember
- [ ] Event
- [ ] Function
- [x] Method
- [ ] Macro
- [x] Keyword
- [ ] Modifier
- [x] Comment
- [ ] String
- [ ] Number
- [ ] Regexp
- [x] Operator
- [ ] Decorator

**Comment by tgodzik** [comment, marked as answer]
> Enum

We could use it for Java Enums with `sym.isJavaEnum` method

> Number

We would need to also add a case for constants when traversing the tree, but it could be used.

> Modifier

I think we have or will have in the next scalameta `ModifierKeyword` trait which could be used for that.

> String

This should be used for String Tokens and String multiline tokens.

Some of the rest we could add in the future, but the above would be great to have now.
